plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java

```diff
@@ -545,26 +543,35 @@ protected IStatus run(IProgressMonitor monitor) {
 								return Status.OK_STATUS;
 							}
 						})
-				.runWhenDone(
-						new Job("Setting connection, deploy project...") {
+				.runWhenDoneIf(job -> { 
+							if (connection == null) {
+								container.getDisplay().syncExec(() -> 
```

Comment:
+1, that's what I did. Upon startup with an opened editor the error message is displayed as expected.
![image](https://cloud.githubusercontent.com/assets/25126/16483400/8806bc2a-3eb4-11e6-8cd4-6a076b1b4579.png)